src/encoded/static/components/navigation/components/LeftNav.js

```diff
@@ -166,3 +171,181 @@ const DataNavItemBody = React.memo(function DataNavItemBody(props) {
+    const [searchInputIsValid, setSearchInputIsValid] = useState(true);
+
+    const onChangeSearchItemType = useCallback(function (evtKey) {
+        if (typeof evtKey === 'string') {
+            setSearchItemType(evtKey);
+
+            //validate accession
+            let isValid = true;
+            if (evtKey === 'ByAccession') {
+                isValid = object.isAccessionRegex(searchText);
+            }
+            if (searchInputIsValid !== isValid) {
+                setSearchInputIsValid(isValid);
+            }
+        }
+    });
+    //render hidden form inputs
+    const renderHiddenInputsForURIQuery = useCallback(function () {
+        const query = {};
+        switch (searchItemType) {
+            case 'ExperimentSetReplicate': {
+                const browseBaseParams = navigate.getBrowseBaseParams();
+                _.extend(query, browseBaseParams);
+            }
+                break;
+            case 'ByAccession':
+                break;
+            default: {
+                _.extend(query, { 'type': searchItemType });
+            }
+                break;
+        }
+        return SearchBar.renderHiddenInputsForURIQuery(query);
+    });
+    //custom placeholder text for search item type
+    const getSearchTextPlaceholder = useCallback(function () {
+        switch (searchItemType) {
+            case 'Item':
+                return 'Search 4DN Data Portal';
+            case 'ByAccession':
+                return 'Type Item\'s Accession (e.g. 4DNXXXX ...)';
+            default:
+                return "Search in " + AvailableSearchItemTypes[searchItemType].text;
+        }
+    });
+    //navigate to Item page directly without searching
+    const navigateByAccession = useCallback(function (evt) {
+        if (searchItemType === 'ByAccession') {
+            evt.preventDefault();
+            evt.stopPropagation();
+            navigate('/' + searchText);
+        }
+    });
+    const handleOnChange = useCallback(function (evt) {
+        const value = evt.target.value;
+        setSearchText(value);
+
+        //validate accession
+        let isValid = true;
+        if (searchItemType === 'ByAccession') {
+            isValid = object.isAccessionRegex(value);
+        }
+        if (searchInputIsValid !== isValid) {
+            setSearchInputIsValid(isValid);
+        }
+    });
+    //select all text when focused
+    const handleFocus = useCallback(function (evt) {
+        evt.target.select();
+    });
+
+    const selectedItem = AvailableSearchItemTypes[searchItemType];
+    const action = (selectedItem && selectedItem.action) || '/search';
+    const btnIconClassName = 'icon icon-fw fas ' + (searchItemType === 'ByAccession' ? 'icon-arrow-right' : 'icon-search');
+    const btnDisabled = !(searchText &&  typeof searchText === 'string' && searchText.length > 0);
+    const searchTextClassName = 'form-control search-query w-100' + (!searchInputIsValid ? ' border border-danger' : '');
```

Comment:
Instead of 'w-100', would get rid of ".form-inline" className from the ancestor `<form>` element. If checkout the CSS definitions, '.form-control' by default is display: block, however form-inline turns form-controls into inline inputs instead of block ones, and thus have to use w-100 or similar to make width stretch, whereas display: block is 100% width by default and would get rid of this issue as well as code/complexity.